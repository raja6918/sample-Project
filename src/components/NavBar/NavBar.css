.MuiIconButton-root {
  padding: 0px;
}
.MuiIconButton-root:hover {
  background-color: rgba(0, 0, 0, 0.08);
}
.profile-content {
  padding: 24px 16px 16px 16px;
}
.heading1,
.heading2 {
  color: #65aeff;
  padding: 24px 20px 12px 24px;
  font-weight: 500;
}
.hamburger-scroll {
  overflow-y: hidden;
}
