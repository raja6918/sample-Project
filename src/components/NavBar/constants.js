export const ScenariosLink = '/scenarios';
export const PairingsLink = '/pairings';
export const SolverLink = '/solver';
export const DataLink = '/data';
export const ReportsLink = '/reports';
export const DashboardLink = '/dashboard';
export const SettingsLink = '/settings';
