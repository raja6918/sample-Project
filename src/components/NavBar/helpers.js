export const amIinScenariosScreen = history =>
  history.location.pathname === '/' ||
  history.location.pathname === '/scenarios';

/**
 * To disable navigation in Navbar and HamburgerDrawers if particular pathname match
 *
 * @param {object} location - from react router dom
 */
export const disableNavigation = location => {
  // If user open pairing in preview mode diable all navigations
  if (location.pathname.includes('pairings-preview')) {
    return true;
  } else if (location.pathname.includes('pairing-details')) {
    return true;
  }
};

/**
 * To enable notification menu if particular pathname match
 *
 * @param {object} location - from react router dom
 */
export const enableNotificationIcon = location => {
  if (location.pathname.includes('pairing-details')) {
    return false;
  }
};
