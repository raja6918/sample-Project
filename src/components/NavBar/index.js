import NavBarComponent from './NavBar';

export default NavBarComponent;
