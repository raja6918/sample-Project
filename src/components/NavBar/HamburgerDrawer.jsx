import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import styled from 'styled-components';
import { Link as RRLink } from 'react-router-dom';

import Drawer from '@material-ui/core/Drawer';
import ListMUI from '@material-ui/core/List';
import ListItemMUI from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import IconButton from '@material-ui/core/IconButton';
import SvgIcon from '@material-ui/core/SvgIcon';
import PerfectScrollbar from 'react-perfect-scrollbar';
import 'react-perfect-scrollbar/dist/css/styles.css';

import HamburguerIcon from '@material-ui/icons/Menu';

import AccessEnabler from '../AccessEnabler';
import history from '../../history';
import storage from '../../utils/storage';
import scopes from '../../constants/scopes';

import Icon from '../Icon';
import IconsPath from './icons/IconsPath';
import './NavBar.css';
import { perfectScrollConfig, currentScenario } from '../../utils/common';

import {
  ScenariosLink,
  PairingsLink,
  SolverLink,
  DataLink,
  ReportsLink,
  DashboardLink,
  SettingsLink,
} from './constants';
import { connect } from 'react-redux';
import { READ_ONLY } from '../../constants';

const HamburgerBTN = styled(IconButton)`
  margin-left: -12px;
  margin-right: 15px;
  width: 48px;
  height: 48px;
  span {
    color: ${props => (props.disabled ? 'inherit' : '#fff')};
  }
  @media (max-width: 576px) {
    margin-right: 0;
  }
`;

const DrawerContent = styled.div`
  width: 320px;
  overflow-x: hidden;
  outline: none;
`;

const Link = styled(RRLink)`
  text-decoration: none;
  width: 100%;
  padding: 12px 24px;
`;

const ProfileContent = styled.div`
  position: relative;
  padding-bottom: 12px;

  & div.profile-content span {
    width: 56px;
    vertical-align: top;
  }

  & div:first-child span p.userName-first {
    margin: 18px auto;
    width: 100%;
    font-size: 15px;
    padding-left: 0px;
  }
  & p {
    margin: 0;
    color: #fff;
    font-size: 16px;
    font-weight: normal;
    padding-left: 15px;
    vertical-align: bottom;
  }
  & span,
  & p {
    display: inline-block;
  }

  & div[role='button'] span,
  & div[role='button'] p {
    display: inline-block;
    vertical-align: middle;
  }
  & div[role='button'] p {
    padding-left: 0;
  }
`;
const CloseChevron = styled.button`
  height: 24px;
  padding: 0;
  border: 0;
  background-color: transparent;
  position: absolute;
  right: -8px;
  top: 12px;
  cursor: pointer;
  & span {
    width: 100%;
    height: 100%;
    margin: 0;
  }
`;

const List = styled(ListMUI)`
  padding: 0;
  & p.heading {
    font-family: Roboto;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    color: #bae1fc;
    padding: 14px 2px 14px 75px;
    max-width: 310px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
  & a {
    display: inline-flex;
    align-items: center;
  }
  & a p {
    font-size: 16px;
    color: #fff;
    margin: 0;
  }
`;

const ListItem = styled(ListItemMUI)`
  padding: ${props => (props.submenu ? '0 0 0 48px' : '0')};
  background-color: ${props => (props.currentSelected ? '#BAE1FC' : 'none')};
  &:hover {
    background: #e8f4fc;
    & p,
    & span {
      color: #0a75c2 !important;
    }
    & path,
    & polygon {
      fill: #0a75c2 !important;
    }
  }
`;

const Ellipsis = styled.p`
  width: 50%;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const DrawerUI = styled(Drawer)`
  color: #fff;
  & > div:nth-child(3n) {
    background-image: linear-gradient(
      0deg,
      #004c8c -2%,
      #0a75c2 1%,
      #012b4f 95%
    );
  }
`;

const MenuItem = ({
  icon,
  link,
  text,
  onClick,
  needState,
  isDisable,
  selected,
  appendResourceId = true,
  submenu,
  ...rest
}) => {
  const openedScenario = storage.getItem('openScenario');
  let route = null;

  if (
    ((history.location && history.location.state) || openedScenario) &&
    needState
  ) {
    const openItemId =
      history.location && history.location.state !== undefined
        ? history.location.state.openItemId
        : openedScenario.id;
    const openItemName =
      history.location.state !== undefined
        ? history.location.state.openItemName
        : openedScenario.name;

    const readOnly =
      history.location.state !== undefined
        ? history.location.state.readOnly
        : openedScenario.status === READ_ONLY
        ? true
        : false;
    const editMode =
      history.location.state !== undefined
        ? history.location.state.editMode
        : !!openedScenario.isTemplate;

    const isScenariosLink = link === ScenariosLink;
    const shouldOpenedScenarioBeUsed = isScenariosLink && !!openedScenario;

    route = {
      pathname: appendResourceId ? `${link}/${openItemId}` : link,
      state: {
        openItemId: shouldOpenedScenarioBeUsed ? openedScenario.id : openItemId,
        openItemName: shouldOpenedScenarioBeUsed
          ? openedScenario.name
          : openItemName,
        editMode,
        readOnly: isScenariosLink ? false : readOnly,
      },
    };
  } else {
    route = link;
  }

  return (
    <ListItem
      currentSelected={selected}
      button
      onClick={onClick}
      disabled={isDisable}
      submenu={!!submenu}
      {...rest}
    >
      <Link to={route}>
        <ListItemIcon style={{ minWidth: '48px' }}>
          <SvgIcon style={{ margin: '0 24px 0 0' }}>
            <IconsPath
              icon={icon}
              color={{ fill: selected ? '#0A75C2' : '#fff' }}
            />
          </SvgIcon>
        </ListItemIcon>
        <p style={{ color: selected ? '#0A75C2' : '#fff' }}>{text}</p>
      </Link>
    </ListItem>
  );
};

export class HamburgerDrawer extends Component {
  state = {
    openDrawer: false,
    showSettings: false,
    enableLine: true,
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.closeNavBar) {
      this.setState({ openDrawer: false, showSettings: false });
    }
  }

  toggleDrawer = () => {
    this.setState({ openDrawer: !this.state.openDrawer, showSettings: false });
  };

  toggleSettings = () => {
    this.setState({ showSettings: !this.state.showSettings });
  };

  createRoute = path => {
    let route = path;
    const openedScenario = storage.getItem('openScenario');
    if (openedScenario) {
      route = {
        pathname: path,
        state: {
          openItemId: openedScenario.id,
          openItemName: openedScenario.name,
          editMode: !!openedScenario.isTemplate,
          readOnly: openedScenario.status === READ_ONLY,
        },
      };
    }
    return route;
  };

  render() {
    const { openDrawer } = this.state;
    const { t, disabled, userPermissions, currentSelected } = this.props;
    const hamburgerMenu = 'NAVBAR.hamburgerMenu';
    const userName = this.props.userName;
    const openScenario = storage.getItem('openScenario');
    const isThereAnItemOpened = openScenario !== null;
    const isDisable =
      typeof history.location.state === 'undefined' && !isThereAnItemOpened;

    const editMode = history.location.state
      ? history.location.state.editMode
      : openScenario && !!openScenario.isTemplate;

    const currentSelectedScenario = currentScenario();

    return (
      <Fragment>
        <HamburgerBTN
          onClick={this.toggleDrawer}
          aria-label="Menu"
          disabled={disabled}
          className="tm-hamburger__menu-btn"
        >
          <HamburguerIcon />
        </HamburgerBTN>
        <DrawerUI open={openDrawer} onClose={this.toggleDrawer}>
          <PerfectScrollbar option={perfectScrollConfig}>
            <DrawerContent
              tabIndex={0}
              role="button"
              className="hamburger-scroll"
            >
              <ProfileContent>
                <div className="profile-content">
                  <CloseChevron onClick={this.toggleDrawer}>
                    <Icon iconcolor={'#fff'}>chevron_left</Icon>
                  </CloseChevron>
                  {/* <div>
                    <Avatar userName={userName} />
                    <Ellipsis>{userName}</Ellipsis>
                  </div> */}
                </div>
                {!!userPermissions.length && (
                  <div className="heading1">MAIN MENU</div>
                )}
                <List component="nav">
                  <AccessEnabler
                    scopes={scopes.pages.dashboard}
                    render={props => (
                      <MenuItem
                        selected={currentSelected === 'dashboard'}
                        icon="dashboardIcon"
                        link={DashboardLink}
                        text={t(`${hamburgerMenu}.dashboard`)}
                        onClick={() => {}}
                        className="tm-hamburger_menu__dashboard-btn"
                        {...props}
                      />
                    )}
                  />
                  <AccessEnabler
                    scopes={scopes.pages.scenario}
                    render={() => (
                      <Fragment>
                        <MenuItem
                          selected={currentSelected === 'scenarios'}
                          icon="scenarioIcon"
                          link={ScenariosLink}
                          text={t(`${hamburgerMenu}.scenarios`)}
                          onClick={() => {
                            this.toggleDrawer();
                          }}
                          needState={isThereAnItemOpened}
                          appendResourceId={false}
                          className="tm-hamburger_menu__scenario-btn"
                        />
                        {currentSelectedScenario &&
                          !currentSelectedScenario.isTemplate && (
                            <List component="nav">
                              <p className="heading">
                                <i>{currentSelectedScenario.name}</i>
                              </p>
                              <MenuItem
                                selected={currentSelected === 'data'}
                                icon="dataIcon"
                                link={DataLink}
                                needState={true}
                                submenu
                                text={t(`${hamburgerMenu}.data`)}
                                onClick={() => {
                                  this.toggleDrawer();
                                }}
                                isDisable={isDisable}
                                className="tm-hamburger_menu__scenario_data-btn"
                              />
                              <MenuItem
                                selected={currentSelected === 'pairing'}
                                icon="pairingsIcon"
                                link={PairingsLink}
                                needState={true}
                                submenu
                                text={t(`${hamburgerMenu}.pairings`)}
                                onClick={() => {
                                  this.toggleDrawer();
                                }}
                                isDisable={!!editMode || isDisable}
                                className="tm-hamburger_menu__scenario_pairings-btn"
                              />
                              <MenuItem
                                selected={currentSelected === 'solver'}
                                icon="solverIcon"
                                submenu
                                link={SolverLink}
                                needState={true}
                                text={t(`${hamburgerMenu}.solver`)}
                                onClick={() => {
                                  this.toggleDrawer();
                                }}
                                isDisable={!!editMode || isDisable}
                                className="tm-hamburger_menu__scenario_solver-btn"
                              />

                              <MenuItem
                                selected={currentSelected === 'reports'}
                                submenu
                                icon="reportsIcon"
                                link={ReportsLink}
                                text={t(`${hamburgerMenu}.reports`)}
                                onClick={() => {}}
                                className="tm-hamburger_menu__scenario_reports-btn"
                              />

                              <MenuItem
                                selected={currentSelected === 'settings'}
                                submenu
                                icon="settingsIcon"
                                link={SettingsLink}
                                text={t('NAVBAR.menuProfile.settings')}
                                onClick={() => {}}
                                className="tm-hamburger_menu__scenario_settings-btn"
                              />
                            </List>
                          )}
                      </Fragment>
                    )}
                  />
                  <AccessEnabler
                    scopes={scopes.pages.templates}
                    render={() => {
                      const style =
                        currentSelected === 'templates'
                          ? { background: '#BAE1FC' }
                          : {};
                      const textStyle =
                        currentSelected === 'templates'
                          ? { color: '#0A75C2' }
                          : {};
                      return (
                        <Fragment>
                          <ListItem
                            style={style}
                            button
                            className="tm-hamburger_menu__templates-btn"
                          >
                            <Link
                              to={this.createRoute('/templates')}
                              onClick={() => {
                                this.toggleDrawer();
                              }}
                            >
                              <Icon
                                iconcolor={
                                  currentSelected === 'templates'
                                    ? '#0A75C2'
                                    : '#fff'
                                }
                                style={{ margin: '0 24px 0 0' }}
                              >
                                library_books
                              </Icon>
                              <p style={textStyle}>
                                {t(`${hamburgerMenu}.templates`)}
                              </p>
                            </Link>
                          </ListItem>
                          {currentSelectedScenario &&
                            currentSelectedScenario.isTemplate && (
                              <List component="nav">
                                <p className="heading">
                                  <i>{currentSelectedScenario.name}</i>
                                </p>
                                <MenuItem
                                  selected={currentSelected === 'data'}
                                  icon="dataIcon"
                                  submenu
                                  link={DataLink}
                                  needState={true}
                                  text={t(`${hamburgerMenu}.data`)}
                                  onClick={() => {
                                    this.toggleDrawer();
                                  }}
                                  isDisable={isDisable}
                                  className="tm-hamburger_menu__templates_data-btn"
                                />
                              </List>
                            )}
                        </Fragment>
                      );
                    }}
                  />
                </List>
                <AccessEnabler
                  scopes={scopes.pages.users}
                  render={() => {
                    const style =
                      currentSelected === 'users'
                        ? { background: '#BAE1FC' }
                        : {};
                    const textStyle =
                      currentSelected === 'users' ? { color: '#0A75C2' } : {};
                    return (
                      <ListItem
                        style={style}
                        button
                        className="tm-hamburger_menu__users-btn"
                      >
                        <Link
                          to={this.createRoute('/users')}
                          onClick={() => {
                            this.toggleDrawer();
                          }}
                        >
                          <Icon
                            iconcolor={
                              currentSelected === 'users' ? '#0A75C2' : '#fff'
                            }
                            style={{ margin: '0 24px 0 0' }}
                          >
                            people
                          </Icon>
                          <p style={textStyle}>{t(`${hamburgerMenu}.users`)}</p>
                        </Link>
                      </ListItem>
                    );
                  }}
                />
              </ProfileContent>
            </DrawerContent>
          </PerfectScrollbar>
        </DrawerUI>
      </Fragment>
    );
  }
}

const mapStatetoProps = state => {
  return {
    notificationsCount: state.notifications.count,
    closeNavBar: state.notifications.closeNavBar,
    userPermissions: state.user.permissions,
    currentSelected: state.notifications.currentNav,
  };
};

MenuItem.propTypes = {
  icon: PropTypes.string.isRequired,
  link: PropTypes.string,
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  needState: PropTypes.bool,
  isDisable: PropTypes.bool,
  appendResourceId: PropTypes.bool,
  closeNavBar: PropTypes.bool,
  selected: PropTypes.bool,
  submenu: PropTypes.bool,
};
MenuItem.defaultProps = {
  onClick: () => {},
  link: '',
  needState: null,
  isDisable: null,
  appendResourceId: true,
  closeNavBar: false,
  selected: false,
  submenu: false,
};
HamburgerDrawer.propTypes = {
  t: PropTypes.func.isRequired,
  userName: PropTypes.string,
  disabled: PropTypes.bool,
  userPermissions: PropTypes.arrayOf(PropTypes.string).isRequired,
  currentSelected: PropTypes.string,
};

HamburgerDrawer.defaultProps = {
  userName: '',
  disabled: false,
  currentSelected: '',
};

const HamburgerDrawerComponent = connect(mapStatetoProps)(HamburgerDrawer);

export default translate()(HamburgerDrawerComponent);
