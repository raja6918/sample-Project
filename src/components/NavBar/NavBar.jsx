import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import styled from 'styled-components';
import { Link as RRLink } from 'react-router-dom';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import Menu from '@material-ui/core/Menu';
import Divider from '@material-ui/core/Divider';
import Icon from '@material-ui/core/Icon';
import Badge from '@material-ui/core/Badge';
import { connect } from 'react-redux';
import * as notificationService from '../../services/Notifications';

import MenuText from '../Menu/MenuText';
import Avatar from '../Avatar';
import HamburgerDrawerComp from './HamburgerDrawer';
import { MenuItemLink } from '../Menu/MenuItem';
import { disableNavigation, enableNotificationIcon } from './helpers';
import './NavBar.css';
import NotificationPanel from '../../components/NotificationPane/NotificationPane';
import storage from '../../utils/storage';
import {
  notificationDecrement,
  updateNotificationAsSeen,
  viewedAllStatus,
  notificationCleanUp,
  closeNotificationBar,
} from '../../actions/generic';
import { currentUserUpdated } from '../../actions/scenario';
import Profile from './UserProfile/Profile';
import upperFirst from 'lodash/upperFirst';
import { userDataCleanUp } from '../../actions/user';
import history from '../../history';

const Root = styled.div`
  width: 100%;
  min-height: 54px;
  height: 100%;
  & header > div {
    background: #004c8c;
    min-height: 54px;
    max-height: 54px;
  }
`;
const Title = styled(Typography)`
  font-family: 'Roboto', sans-serif;
  font-size: 24px;
  color: #fff;
  a {
    text-decoration: none;
    color: inherit;
  }
  @media (max-width: 576px) {
    display: none;
  }
`;
const ContainerBTNs = styled.div`
  flex: 1;
  div {
    display: flex;
    float: right;
  }
`;
const NotificationsBTN = styled(IconButton)`
  margin-right: 10px;
  color: #fff;
  width: 48px;
  height: 48px;
  @media (max-width: 576px) {
    margin-right: 0;
  }
`;
const ProfileBTN = styled(IconButton)`
  width: auto;
  color: #fff;
  border-radius: 0;
  padding-right: 0;
`;
const MenuProfile = styled(Menu)`
  top: 34px;
  & .MuiPaper-root {
    height: 171px;
    min-width: 336px;
    max-width: 433px;
    overflow: hidden;
    border-radius: 3px;
    box-shadow: 0 0 24px 0 rgb(0 0 0 / 35%);
  }
`;
const Link = styled(RRLink)`
  width: 100%;
  height: 100%;
  display: inline-flex;
  padding: 20px;
  text-decoration: none;
`;

export class NavBar extends Component {
  state = {
    anchorEl: null,
    isNotificationsOpen: false,
    firstName: '',
    lastName: '',
    role: '',
  };

  disableNavigation = false;
  enableNotificationIcon = false;
  componentDidMount() {
    const { firstName, lastName, role } = this.props.loggedUser;
    this.setState({ firstName, lastName, role });
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.closeNavBar && this.state.isNotificationsOpen) {
      this.toggleNotificationPane();
    }
    if (nextProps.currentUserUpdated) {
      const { firstName, lastName, role } = storage.getItem('loggedUser');
      this.setState({ firstName, lastName, role }, () => {
        this.props.resetCurrentUserUpdated(false);
      });
    }
  }
  handleMenu = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };
  notificationViewed = () => {
    this.props.notificationViewed();
  };

  getRoute = () => {
    const { readOnly } = this.props;
    const openedScenario = storage.getItem('openScenario');
    let route = null;

    if ((history.location && history.location.state) || openedScenario) {
      const openItemId =
        history.location && history.location.state
          ? history.location.state.openItemId
          : openedScenario.id;
      const openItemName =
        history.location && history.location.state
          ? history.location.state.openItemName
          : openedScenario.name;

      route = {
        pathname: '/',
        state: {
          openItemId: openedScenario ? openedScenario.id : openItemId,
          openItemName: openedScenario ? openedScenario.name : openItemName,
          editMode: openedScenario ? openedScenario.isTemplate : false,
          readOnly,
        },
      };
    } else {
      route = '/';
    }

    return route;
  };

  toggleNotificationPane = () => {
    this.setState(
      prevState => ({
        isNotificationsOpen: !prevState.isNotificationsOpen,
      }),
      () => {
        if (this.state.isNotificationsOpen) {
          const { loggedUser } = this.props;
          const payload = {
            status: 'SEEN',
          };
          notificationService
            .updateAllNotifications(payload, loggedUser.id)
            .then(() => {
              this.props.viewedAllStatus();
            })
            .catch(error => {
              console.error(error);
            });
        } else {
          this.props.updateNotificationAsSeen();
          if (this.props.notificationsCount) this.props.notificationViewed();
          this.props.closeNotificationBar(false);
        }
      }
    );
  };

  flushAllData = () => {
    const { notificationCleanUp, userDataCleanUp } = this.props;
    notificationCleanUp();
    userDataCleanUp();
  };

  render() {
    const { t, currentPath } = this.props;
    const {
      anchorEl,
      isNotificationsOpen,
      firstName,
      lastName,
      role,
    } = this.state;
    const open = Boolean(anchorEl);
    const menuProfile = 'NAVBAR.menuProfile';
    const fName = upperFirst(firstName);
    const lName = upperFirst(lastName);
    const userName = `fName lName`;
    this.disableNavigation = disableNavigation(location);
    this.enableNotificationIcon = enableNotificationIcon(location);
    return (
      <Root>
        <AppBar position="fixed">
          <Toolbar>
            <HamburgerDrawerComp
              t={t}
              userName={userName}
              disabled={this.disableNavigation}
              currentPath={currentPath}
            />
            <Title variant="title" color="inherit">
              {this.disableNavigation ? (
                <span>SIERRA</span>
              ) : (
                <RRLink to={this.getRoute()}>SIERRA</RRLink>
              )}
            </Title>
            {/*<SearchEngine />*/}
            <ContainerBTNs>
              <div>
                <NotificationsBTN
                  disabled={
                    this.disableNavigation && !this.enableNotificationIcon
                  }
                  onClick={this.toggleNotificationPane}
                >
                  <Badge
                    badgeContent={this.props.notificationsCount}
                    color="secondary"
                  >
                    <Icon>notifications</Icon>
                  </Badge>
                </NotificationsBTN>
                <ProfileBTN
                  aria-owns={open ? 'menu-appbar' : null}
                  aria-haspopup="true"
                  onClick={this.handleMenu}
                  color="inherit"
                  disabled={this.disableNavigation}
                >
                  <Avatar
                    firstName={fName}
                    lastName={lName}
                    disabled={this.disableNavigation}
                  />
                  <Icon>arrow_drop_down</Icon>
                </ProfileBTN>
                <MenuProfile
                  id="menu-appbar"
                  anchorEl={anchorEl}
                  getContentAnchorEl={null}
                  anchorOrigin={{
                    vertical: 'bottom',
                    horizontal: 'right',
                  }}
                  transformOrigin={{
                    vertical: 'top',
                    horizontal: 'right',
                  }}
                  open={open}
                  onClose={this.handleClose}
                >
                  <Profile
                    firstName={fName}
                    lastName={lName}
                    userRole={role}
                    disabled={this.disableNavigation}
                  />
                  {/* <MenuItem onClick={this.handleClose}>
                    <MenuIcon>person</MenuIcon>
                    <MenuText>{t(`${menuProfile}.profile`)}</MenuText>
                  </MenuItem> */}
                  {/* <MenuItem onClick={this.handleClose}>
                    <MenuIcon>settings</MenuIcon>
                    <MenuText>{t(`${menuProfile}.settings`)}</MenuText>
                  </MenuItem> */}
                  <Divider />
                  <MenuItemLink onClick={this.handleClose}>
                    <Link to="/logout" onClick={this.flushAllData}>
                      <MenuText
                        style={{ opacity: 0.9, fontWeight: 400 }}
                        textColor="#333333"
                      >
                        {t(`${menuProfile}.signOut`)}
                      </MenuText>
                    </Link>
                  </MenuItemLink>
                </MenuProfile>
              </div>
            </ContainerBTNs>
            <NotificationPanel
              t={t}
              isOpen={isNotificationsOpen}
              handleCancel={this.toggleNotificationPane}
            />
          </Toolbar>
        </AppBar>
      </Root>
    );
  }
}

const mapStatetoProps = state => {
  return {
    notificationsCount: state.notifications.count,
    closeNavBar: state.notifications.closeNavBar,
    currentUserUpdated: state.scenario.currentUserUpdated,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    notificationViewed: () => dispatch(notificationDecrement()),
    updateNotificationAsSeen: () => dispatch(updateNotificationAsSeen()),
    viewedAllStatus: () => dispatch(viewedAllStatus()),
    notificationCleanUp: () => dispatch(notificationCleanUp()),
    closeNotificationBar: () => dispatch(closeNotificationBar()),
    resetCurrentUserUpdated: bool => dispatch(currentUserUpdated(bool)),
    userDataCleanUp: () => dispatch(userDataCleanUp()),
  };
};

NavBar.propTypes = {
  t: PropTypes.func.isRequired,
  loggedUser: PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    email: PropTypes.string,
    role: PropTypes.string,
    userName: PropTypes.string,
  }),
  notificationsCount: PropTypes.number,
  notificationViewed: PropTypes.func.isRequired,
  updateNotificationAsSeen: PropTypes.func.isRequired,
  viewedAllStatus: PropTypes.func.isRequired,
  notificationCleanUp: PropTypes.func.isRequired,
  closeNavBar: PropTypes.bool.isRequired,
  closeNotificationBar: PropTypes.func.isRequired,
  currentUserUpdated: PropTypes.bool.isRequired,
  resetCurrentUserUpdated: PropTypes.func.isRequired,
  userDataCleanUp: PropTypes.func.isRequired,
  readOnly: PropTypes.bool,
};

NavBar.defaultProps = {
  loggedUser: {},
  notificationsCount: 0,
  readOnly: false,
};
const NavComponent = connect(mapStatetoProps, mapDispatchToProps)(NavBar);
export default translate()(NavComponent);
