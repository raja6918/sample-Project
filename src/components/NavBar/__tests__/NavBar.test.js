import React from 'react';
import { shallow } from 'enzyme';
import { NavBar } from '../NavBar';

const minProps = {
  t: jest.fn(),
  loggedUser: {
    id: 1,
    firstName: 'test',
    lastName: 'test',
    email: 'test@test.com',
    role: 'Planner',
    userName: 'test',
  },
  notificationsCount: 0,
  notificationViewed: jest.fn(),
  updateNotificationAsSeen: jest.fn(),
  viewedAllStatus: jest.fn(),
  notificationCleanUp: jest.fn(),
  closeNavBar: false,
  closeNotificationBar: jest.fn(),
};

let wrapper;
describe(`Test Suite for Navbar`, () => {
  beforeEach(() => {
    wrapper = shallow(<NavBar {...minProps} />);
  });

  test(`NavBar component renders`, () => {
    expect(wrapper).toMatchSnapshot();
  });

  test(`check handleMenu works as expected`, () => {
    expect(wrapper.state('anchorEl')).toBeNull();
    wrapper.instance().handleMenu({ currentTarget: 'test' });
    expect(wrapper.state('anchorEl')).toBe('test');
  });

  test(`check handleClose works as expected`, () => {
    expect(wrapper.state('anchorEl')).toBeNull();
    wrapper.instance().handleMenu({ currentTarget: 'test' });
    expect(wrapper.state('anchorEl')).toBe('test');
    wrapper.instance().handleClose();
    expect(wrapper.state('anchorEl')).toBeNull();
  });

  test(`check getRoute return exepected output`, () => {
    expect(wrapper.instance().getRoute()).toBe('/');
  });

  test(`check toggleNotificationPane works as expected`, () => {
    wrapper.instance().toggleNotificationPane();
    expect(wrapper.state('isNotificationsOpen')).toBeTruthy();
    wrapper.instance().toggleNotificationPane();
    expect(wrapper.state('isNotificationsOpen')).toBeFalsy();
  });
});
