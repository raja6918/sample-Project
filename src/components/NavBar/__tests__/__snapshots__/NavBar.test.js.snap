// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Suite for Navbar NavBar component renders 1`] = `
<NavBar__Root>
  <WithStyles(ForwardRef(AppBar))
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <Component
        t={
          [MockFunction] {
            "calls": Array [
              Array [
                "NAVBAR.menuProfile.signOut",
              ],
              Array [
                "NAVBAR.menuProfile.signOut",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        userName="fName lName"
      />
      <NavBar__Title
        color="inherit"
        variant="title"
      >
        <Link
          replace={false}
          to="/"
        >
          SIERRA
        </Link>
      </NavBar__Title>
      <NavBar__ContainerBTNs>
        <div>
          <NavBar__NotificationsBTN
            onClick={[Function]}
          >
            <WithStyles(ForwardRef(Badge))
              badgeContent={0}
              color="secondary"
            >
              <WithStyles(ForwardRef(Icon))>
                notifications
              </WithStyles(ForwardRef(Icon))>
            </WithStyles(ForwardRef(Badge))>
          </NavBar__NotificationsBTN>
          <NavBar__ProfileBTN
            aria-haspopup="true"
            aria-owns={null}
            color="inherit"
            onClick={[Function]}
          >
            <ProfileIcon
              disabled={false}
              firstName="Test"
              lastName="Test"
            />
            <WithStyles(ForwardRef(Icon))>
              arrow_drop_down
            </WithStyles(ForwardRef(Icon))>
          </NavBar__ProfileBTN>
          <NavBar__MenuProfile
            anchorEl={null}
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "bottom",
              }
            }
            getContentAnchorEl={null}
            id="menu-appbar"
            onClose={[Function]}
            open={false}
            transformOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
          >
            <Profile
              disabled={false}
              firstName="Test"
              lastName="Test"
              userRole="Planner"
            />
            <WithStyles(ForwardRef(Divider)) />
            <MenuItem__MenuItemLink
              onClick={[Function]}
            >
              <NavBar__Link
                onClick={[Function]}
                to="/logout"
              >
                <MenuText
                  style={
                    Object {
                      "fontWeight": 400,
                      "opacity": 0.9,
                    }
                  }
                  textColor="#333333"
                />
              </NavBar__Link>
            </MenuItem__MenuItemLink>
          </NavBar__MenuProfile>
        </div>
      </NavBar__ContainerBTNs>
      <Connect(NotificationPane)
        handleCancel={[Function]}
        isOpen={false}
        t={
          [MockFunction] {
            "calls": Array [
              Array [
                "NAVBAR.menuProfile.signOut",
              ],
              Array [
                "NAVBAR.menuProfile.signOut",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</NavBar__Root>
`;
